Mathlib/LinearAlgebra/CliffordAlgebra/SpinGroup.lean

```diff
@@ -73,127 +73,110 @@ def inv_of_inv_ι (m : M) [Invertible (ι Q m)] [Invertible (2 : R)] : Invertibl
 
 #align invertible_of_invertible_ι inv_of_inv_ι
 
--- TODO: below are not fixed yet
+-- restore the lean3 behavior
+macro_rules | `($x * $y) => `(@HMul.hMul _ _ _ instHMul $x $y)
 
 /-- `lipschitz` is the subgroup closure of all the elements in the form of `ι Q m` where `ι`
 is the canonical linear map `M →ₗ[R] CliffordAlgebra Q`. -/
 def lipschitz (Q : QuadraticForm R M) :=
-  Subgroup.closure (coe ⁻¹' Set.range (ι Q) : Set (CliffordAlgebra Q)ˣ)
+  Subgroup.closure (Units.val ⁻¹' Set.range (ι Q) : Set (CliffordAlgebra Q)ˣ)
 #align lipschitz lipschitz
 
+set_option synthInstance.maxHeartbeats 25000 in
 /-- If x is in `lipschitz Q`, then `(ι Q).range` is closed under twisted conjugation. The reverse
 statement presumably being true only in finite dimensions.-/
 theorem mem_lipschitz_conjAct_le {x : (CliffordAlgebra Q)ˣ} [Invertible (2 : R)]
-    (hx : x ∈ lipschitz Q) : ConjAct.toConjAct x • (ι Q).range ≤ (ι Q).range :=
-  by
-  refine' @Subgroup.closure_induction'' _ _ _ _ _ hx _ _ _ _
-  · rintro x ⟨z, hz⟩ y ⟨a, ha⟩
-    simp only [SMul.smul, SetLike.mem_coe, LinearMap.mem_range, DistribMulAction.toLinearMap_apply,
-      ConjAct.ofConjAct_toConjAct] at ha
-    rcases ha with ⟨⟨b, hb⟩, ha1⟩
+    (hx : x ∈ lipschitz Q) : ConjAct.toConjAct x • LinearMap.range (ι Q) ≤ LinearMap.range (ι Q) := by
+  unfold lipschitz at hx
+  apply Subgroup.closure_induction'' hx
+  · rintro x ⟨a, ha⟩ y ⟨z, ⟨⟨b, hb⟩, hz⟩⟩
+    have := x.invertible
+    have : Invertible (ι Q a) := by rwa [ha]
+    have : Invertible (Q a) := inv_of_inv_ι a
+    rw [LinearMap.mem_range]
+    simp only [HSMul.hSMul, SMul.smul, DistribMulAction.toLinearMap_apply, ConjAct.ofConjAct_toConjAct,
+                SetLike.mem_coe, LinearMap.mem_range] at hz -- diff
     subst hb
-    letI := x.invertible
-    letI : Invertible (ι Q z) := by rwa [hz]
-    rw [LinearMap.mem_range, ← ha1, ← invOf_units x]
-    suffices ∃ y : M, (ι Q) y = ι Q z * (ι Q) b * ⅟ (ι Q z)
-      by
-      convert this
-      ext1
-      congr <;> simp only [hz.symm, Subsingleton.helim (congr_arg Invertible hz.symm)]
-    letI := invertibleOfInvertibleι Q z
-    refine' ⟨(⅟ (Q z) * QuadraticForm.polar Q z b) • z - b, (ι_mul_ι_mul_inv_of_ι Q z b).symm⟩
-  · rintro x ⟨z, hz1⟩ y ⟨a, ⟨b, hb⟩, ha2⟩
-    simp only [ConjAct.toConjAct_inv, DistribMulAction.toLinearMap_apply, SMul.smul,
-      ConjAct.ofConjAct_inv, ConjAct.ofConjAct_toConjAct, inv_inv] at ha2
+    suffices ∃ y : M, ι Q y = ι Q a * ι Q b * ⅟ (ι Q a) by simp_all only [invOf_units]
+    rw [ι_mul_ι_mul_invOf_ι Q a b]
+    use ((⅟ (Q a) * QuadraticForm.polar Q a b) • a - b)
```

Comment:
I wonder how can I prove this without an explicit `use`, I have no way of knowing the closed form of that vector.

Maybe some lemmas should be added back to be near `ι_mul_ι_mul_invOf_ι` and alike to avoid them here.